### PR TITLE
fix(inspect): use quoting style from stylua.toml

### PR DESCRIPTION
**Pull request by clason**
Problem: `vim.inspect` uses double quotes by default, which means that
copying and pasting the output into our sources will fail CI.

Solution: Flip quotes.

Note: The output still is not guaranteed to pass linting (trailing
commas, line lengths), but this reduces the number of changes required.

### CODE REVIEW

File: runtime/lua/vim/_meta/options.lua [hunk 1]
```diff
@@ -1926,7 +1926,7 @@ vim.go.ef = vim.go.errorfile
...
+vim.o.errorformat = '%*[^"]"%f"%*\\D%l: %m,"%f"%*\\D%l: %m,%-Gg%\\?make[%*\\d]: *** [%f:%l:%m,%-Gg%\\?make: *** [%f:%l:%m,%-G%f:%l: (Each undeclared identifier is reported only once,%-G%f:%l: for each function it appears in.),%-GIn file included from %f:%l:%c:,%-GIn file included from %f:%l:%c\\,,%-GIn file included from %f:%l:%c,%-GIn file included from %f:%l,%-G%*[ ]from %f:%l:%c,%-G%*[ ]from %f:%l:,%-G%*[ ]from %f:%l\\,,%-G%*[ ]from %f:%l,%f:%l:%c:%m,%f(%l):%m,%f:%l:%m,"%f"\\, line %l%*\\D%c%*[^ ] %m,%D%*\\a[%*\\d]: Entering directory %*[`\']%f\',%X%*\\a[%*\\d]: Leaving directory %*[`\']%f\',%D%*\\a: Entering directory %*[`\']%f\',%X%*\\a: Leaving directory %*[`\']%f\',%DMaking %*\\a in %f,%f|%l| %m'
```
Comment (clason) [inline runtime/lua/vim/_meta/options.lua:1929]: 👀 

(Note that `stylua` wants to format this differently due to the `AutoPrefer` option, which `inspect.lua` is not smart enough for in this case.)

**Comment by clason** [comment]
Not sure if the meta and test churn is worth it?

**Review by justinmk — APPROVED** [review]
should `runtime/lua/vim/_meta/options.lua` be removed from an ignore list somewhere?

**Comment by clason** [comment]
No, since the `vim.inspect` output still doesn't pass lualint.

**Comment by clason** [comment]
Don't know why the `ui_spec` is failing (and what it expects).

**Comment by zeertzjq** [comment]
Why should the behavior of an stdlib function depend on Nvim source code conventions?

File: test/functional/lua/ui_spec.lua [hunk 2]
```diff
@@ -142,7 +142,7 @@ describe('vim.ui', function()
       end
       if not is_os('bsd') then
         matches(
-          'vim.ui.open: command failed %(%d%): { "[^"]+", .*"non%-existent%-file" }',
+          "vim.ui.open: command failed %(%d%): { '[^']+', 'non%-existent%-file' }",
```
Comment (zeertzjq) [inline test/functional/lua/ui_spec.lua:145]:
```suggestion
          "vim.ui.open: command failed %(%d%): { '[^']+', .*'non%-existent%-file' }",
```

**Comment by justinmk** [comment]
> Why should the behavior of an stdlib function depend on Nvim source code conventions?

We have to choose something, so why not align with the conventions we intentionally selected?

- Eliminates some contributor confusion and toil. 
- We chose the conventions because we think they are good. So it makes sense for our stdlib to produce things that are also good.

**Pull request closed without merging**